validation.go

```diff
@@ -1,5 +1,28 @@
 package fyne
 
+// ExpandedValidator expands on the Validatable interface to allow the widget to expose more context to the parent.
+//
+// Since: 2.5
+type ExpandedValidator interface {
+	Validatable
+
+	// HasValidator makes it possible for a parent that cares about child validation (e.g. widget.Form)
+	// to know if the developer has attached a validator or not.
+	HasValidator() bool
+
+	// SetOnFocusChanged is intended for parent widgets or containers to hook into the change of focus.
+	// The function might be overwritten by a parent that cares about child validation (e.g. widget.Form).
+	SetOnFocusChanged(func(bool))
+
+	// SetValidationError manually updates the validation status until the next input change.
+	SetValidationError(error)
```

Comment:
I'm not sure about this either - it means that validation moves from a action to a state.
I would look carefully at where it is used now - I think that the new methods added should remove the necessity to set an initial state, especially if we have interception of focus events.